src/blockchain/chain.rs

```diff
@@ -42,19 +42,43 @@ impl Chain {
         Self { blocks }
     }
 
+    /// This function checks if all blocks in the chain are valid by comparing each block with its predecessor.
```

Comment:
I think this should be on top if `is_chain_valid`, correct?

Reply:
Ah no, this seems like a copy/paste error

Reply:
yes, i miss up the docs for this function